src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -1170,4 +1172,116 @@ private function addRobotsIndexSection(ArrayNodeDefinition $rootNode)
             ->end()
         ;
     }
+
+    private function addHttpClientSection(ArrayNodeDefinition $rootNode)
+    {
+        $subNode = $rootNode
+            ->children()
+                ->arrayNode('http_client')
+                    ->info('HTTP Client configuration')
+                    ->canBeEnabled()
+                    ->fixXmlConfig('client')
+                    ->children();
+        $subNode = $this->addHttpClientOptionsSection($subNode);
+        $subNode = $subNode
+                        ->arrayNode('clients')
+                            ->useAttributeAsKey('name')
+                            ->normalizeKeys(false)
+                            ->arrayPrototype()
+                                ->children();
+        $subNode = $this->addHttpClientOptionsSection($subNode);
+        $subNode = $subNode
+                                ->end()
+                            ->end()
+                        ->end()
+                    ->end()
+                ->end()
+            ->end()
+        ;
+    }
+
+    private function addHttpClientOptionsSection(NodeBuilder $rootNode): NodeBuilder
```

Comment:
I wouldn't return a node builder here. This just makes it harder to follow the other tree modifications in `addHttpClientSection()`.

Reply:
fixed